tests/CollectionTest.php

```diff
@@ -2062,6 +2066,77 @@ public function testIsVersioningEnabled()
         ));
         $this->assertFalse($this->database->col3->isVersioningEnabled());
     }
+
+    /**
+     * @throws Exception
+     */
+    public function testRenameNonExistentCollection()
+    {
+        $this->expectException(Exception::class);
+        $this->expectExceptionCode(26);
+        $this->expectExceptionMessage('Error: #26: source namespace does not exist');
+
+        // set test collection
+        $collection = $this->database->getCollection('test');
+
+        // rename non-existent collection to cause a MongoClient error
+        $collection->renameCollection('test.phpmongo_test_collection');
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testReplaceCollection()
+    {
+        $sourceCollection = 'source_collection';
+
+        // creating fake-collection to rename it then in test db context
+        $this->database->createCollection($sourceCollection);
+
+        $collection = $this->database->getCollection($sourceCollection);
+
+        // rename an existent collection
+        $newCollection = $collection->replaceCollection('test.phpmongo_test_collection');
+
+        $this->assertInstanceOf(Collection::class, $newCollection);
+        $this->assertEquals($newCollection->getDatabase()->getName(), 'test');
+        $this->assertEquals($newCollection->getName(), 'phpmongo_test_collection');
+
+        $document = $newCollection
+            ->createDocument(array('param' => 'value'))
+            ->save();
+
+        $this->assertEquals('value', $document->param);
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testRenameCollection()
+    {
+        $sourceCollection = 'source_collection';
+
+        // creating fake-collection to rename it then in test db context
+        $this->database->createCollection($sourceCollection);
+
+        $collection = $this->database->getCollection($sourceCollection);
+
+        // del to rename fake one
+        $this->collection->delete();
+
+        // rename an existent collection
+        $newCollection = $collection->renameCollection('test.phpmongo_test_collection');
```

Comment:
also required test where you try to rename to existed collection and get exception